client/state/jetpack-connect/reducer.js

```diff
@@ -264,6 +265,29 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 	return state;
 }
 
+export function jetpackAuthAttempts( state = {}, action ) {
+	switch ( action.type ) {
+		case JETPACK_CONNECT_RETRY_AUTH:
+			const slug = action.slug;
+			let currentTimestamp = state[ slug ] ? state[ slug ].timestamp || Date.now() : Date.now();
+			let attemptNumber = action.attemptNumber;
+			if ( attemptNumber > 0 ) {
+				const now = Date.now();
+				if ( now - currentTimestamp > AUTH_ATTEMPS_TTL ) {
+					currentTimestamp = now;
+					attemptNumber = 0;
+				}
+			}
+			return Object.assign( {}, state, { [ slug ]: { attempt: attemptNumber, timestamp: currentTimestamp } } );
+		case JETPACK_CONNECT_COMPLETE_FLOW:
+			return {};
+		case DESERIALIZE:
```

Comment:
I think we should be deleting stale data here as well. So if we integrate `isStale` here, we will not need the `attemptNumber > 0` logic, and this can simplify the reducer code for `JETPACK_CONNECT_RETRY_AUTH`.